### ISSUE TITLE
Microk8s Cluster MACVLAN IPAM Assigns Duplicate IP addresses for 6 Pod Deployment

### ISSUE THREAD

**Issue by dominickcafarelli** [canonical/microk8s#3008]
Please run `microk8s inspect` and attach the generated tarball to this issue.

We appreciate your feedback. Thank you for using microk8s.

Hello Team,

I have setup a 3 node cluster. Each with the same physical network configuration.
I have setup a Multus MACVLAN network attachment with IPAM set to have IP Addresses assigned from a pool of 20 IP Addresses.  Assume that the IP network is 192.168.1.1/24 and the IP address range for the IPAM is 192.168.1.40 through 192.168.1.59
All three nodes have the same network interface attached to a physical network supporting this range of IP Addresses on 192.168.1.1/24
I have setup a POD deployment where one of the POD network interfaces is instructed to use this IP Address range. With replica set of 6.  

6 Pods are successfully created where 2 pods are on each of the 3 nodes in the Microk8s cluster.

As I review the IP Addresses assigned to the multus MACVLAN interface on each of these pods, I would expect each pod in the deployment to receive a unique IP address in the range of 192.168.1.40 through 192.168.1.59.

However what I find is that IP Addresses are duplicated across the 6 nodes. Of course each node has two pods and those IP addresses are not duplicated on the same node.  

See the following Table:
Pod 1 - Node 1 - IP 192.168.1.40
Pod 2 - Node 1 - IP 192.168.1.41
Pod 3 - Node 2 - IP 192.168.1.41
Pod 4 - Node 2 - IP 192.168.1.40
Pod 5 - Node 3- IP 192.168.1.40
Pod 6 - Node 3 - IP 192.168.1.41

I also tried to use the IPAM Type to be dhcp and that did not work for a different reason.

Is there a way in the Deployment YAML or the MULTUS YAML to add an option that will ensure that each POD in the deployment will have a unique IP Address on its network.

I plan to use these IP Addresses for bidirectional flows.

Thank you,
Dominick

**Comment by siva-user-fr** [comment]
Please try to use IPAM Type: whereabouts is so straight forward will assign IP addresses cluster-wide.

https://github.com/k8snetworkplumbingwg/whereabouts

We had a similar issue in our multi-cluster and resolved with whereabouts IPAM Plugin.

**Comment by dominickcafarelli** [comment]
> Please try to use IPAM Type: whereabouts is so straight forward will assign IP addresses cluster-wide.
> 
> https://github.com/k8snetworkplumbingwg/whereabouts
> 
> We had a similar issue in our multi-cluster and resolved with whereabouts IPAM Plugin.

Thank you for the suggestion and I will implement accordingly. Of course this makes sense and I figured that the word "host-local" meant that the host pool would be used. I tried defining a different pool on each host but that does not work in the cluster mode (or maybe I just could not figure out how to do that). And the description of the whereabouts plugin makes perfect sense that the allocation is managed at the cluster level. I guess I assumed that this whereabouts plugin would be included in the Microk8s when enabling multus but it is not. Especially since multus + cluster basically requires such an IPAM model.
